cartographer/resources/resource_registry.py

```diff
@@ -28,11 +29,13 @@ def call_initialization_hook(self):
 
     def register_resource(self, type_string, schema, model=None,
                           serializer=None, parser=None, mask=None):
-        self.registry[type_string] = filter_dict({
+        existing_registration = self.registry[type_string]
+        additional_details = filter_dict({
             ResourceRegistryKeys.TYPE: type_string,
             ResourceRegistryKeys.SCHEMA: schema,
             ResourceRegistryKeys.MODEL: model,
             ResourceRegistryKeys.SERIALIZER: serializer,
             ResourceRegistryKeys.PARSER: parser,
             ResourceRegistryKeys.MASK: mask,
         })
+        self.registry[type_string] = dict(existing_registration.items() | additional_details.items())
```

Comment:
I believe that you can just do an `.update` because you already have a `defaultdict`!

```self.registry[type_string].update(additional_details)```

Then assigning to `existing_registration` is unnecessary.